lib/cdo/geocoder.rb

```diff
@@ -116,10 +120,11 @@ def self.find_potential_street_address(text)
     results = Geocoder.search(first_number_to_end)
     return nil if results.empty?
 
-    if results.first&.street_address
-      return first_number_to_end
-    end
-    nil
+    # Return nil if none of the results returned from Geocoder matched on a
+    # street address with relevance >= 0.8
+    return nil if results.none? {|r| r.relevance >= 0.8 && r.street_address}
```

Comment:
0.8 is a number that I picked somewhat arbitrarily.  From my experiments, "good" matches typically have relevance == 1 or very close and "bad" matches look like they have relevance < 0.6.